lib/BoundaryValueDiffEqShooting/src/single_shooting.jl

```diff
@@ -119,27 +117,30 @@ function __single_shooting_loss(u, p, cache, bc::BC, u0_size, pt) where {BC}
 end
 
 function __single_shooting_jacobian!(J, u, jac_cache, diffmode, loss_fn::L, fu) where {L}
-    sparse_jacobian!(J, diffmode, jac_cache, loss_fn, fu, vec(u))
+    DI.jacobian!(loss_fn, fu, J, jac_cache, diffmode, vec(u))
     return J
 end
 
 function __single_shooting_jacobian(J, u, jac_cache, diffmode, loss_fn::L) where {L}
-    sparse_jacobian!(J, diffmode, jac_cache, loss_fn, vec(u))
+    DI.jacobian!(loss_fn, J, jac_cache, diffmode, vec(u))
     return J
 end
 
 function __single_shooting_jacobian_ode_cache(
-        prob, jac_cache, ::NoDiffCacheNeeded, u0, ode_alg; kwargs...)
+        prob, jac_cache, ::NoDiffCacheNeeded, diffmode, u0, ode_alg; kwargs...)
     return SciMLBase.__init(remake(prob; u0), ode_alg; kwargs...)
 end
 
 function __single_shooting_jacobian_ode_cache(
-        prob, jac_cache, ::DiffCacheNeeded, u0, ode_alg; kwargs...)
-    cache = jac_cache.cache
-    if cache isa ForwardDiff.JacobianConfig
-        xduals = cache.duals isa Tuple ? cache.duals[2] : cache.duals
+        prob, jac_cache, ::DiffCacheNeeded, diffmode, u0, ode_alg; kwargs...)
+    if diffmode isa AutoSparse
+        T_dual = eltype(overloaded_input_type(jac_cache))
+        xduals = zeros(T_dual, size(u0))
+    elseif diffmode isa AutoForwardDiff
+        T_dual = eltype(overloaded_input_type(jac_cache))
+        xduals = zeros(T_dual, size(u0))
     else
-        xduals = cache.t
+        xduals = reshape(cache.t[1:length(u0)], size(u0))
```

Comment:
Same remark on the use of `cache.t` (`cache` is no longer defined and `t` is an internal field